sklearn/calibration.py

```diff
@@ -266,12 +279,13 @@ def fit(self, X, y, sample_weight=None):
 
             cv = check_cv(self.cv, y, classifier=True)
             fit_parameters = signature(base_estimator.fit).parameters
-            supports_sw = "sample_weight" in fit_parameters
-
+            base_estimator_supports_sw = "sample_weight" in fit_parameters
+            base_estimator_uses_sw = (
+                sample_weight is not None and base_estimator_supports_sw)
             if sample_weight is not None:
                 sample_weight = _check_sample_weight(sample_weight, X)
 
-                if not supports_sw:
+                if not base_estimator_supports_sw:
```

Comment:
This is a bit weird here because we are rechecking for th outer condition

```python
if not base_estimator_supports_sw:
```

is enough

Reply:
Do I understand correctly that you would like to move up the `if not base_estimator_supports_sw` check outside of `if sample_weight is not None`? If so, we would warn about not using `sample_weight` in the base estimator even if `sample_weight is None`, which I think would be unnecessary.